tiledb/api/src/error.rs

```diff
@@ -28,6 +28,8 @@ impl Drop for RawError {
 
 #[derive(Clone, Debug)]
 pub enum DatatypeErrorKind {
+    ExpectedFixedSize,
+    ExpectedVarSize,
```

Comment:
Add some fields for context? The naem of the attribute would be helpful if it is available, but definitely at least it could have the cell val num that was actually found

Reply:
Fair point. Will do.

Reply:
Turns out I didn't even end up using these so I've removed them. I keep wanting them, but I've also been figuring out ways of needing them. Not entirely sure what that means.